src/main/java/dev/mlml/command/Context.java

```diff
@@ -76,12 +90,15 @@ public boolean parse(Command command) {
                 parsedArguments.add(arg, String.join(" ", List.of(args).subList(i + offset, args.length)));
                 break;
             }
-            if (arg.isRequired() && i > args.length + i + offset) {
-                logger.debug("Required argument not found: {}, args.length: {}, offset: {}",
-                             arg.getName(),
-                             args.length,
-                             offset
-                );
+            if (i > args.length + i + offset) {
+                if (arg.isRequired()) {
+                    logger.debug("Required argument not found: {}, args.length: {}, offset: {}",
+                                 arg.getName(),
+                                 args.length,
+                                 offset
+                    );
+                    parsedArguments.invalidate();
+                }
                 return true;
```

Comment:
Not entirely convinced this works in the case that it seeks past option arguments, does it fill all of the remaining ones so they don't return null before returning?